src/reducers/fileTree.spec.tsx

```diff
@@ -1012,4 +1013,125 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('findRelativePathWithDiff', () => {
+    const file1 = 'file1.js';
+    const file2 = 'file2.js';
+    const file3 = 'file3.js';
+
+    const entries = [
+      createInternalVersionEntry({
+        ...fakeVersionEntry,
+        filename: file1,
+        mime_category: 'text',
+        path: file1,
+        status: 'M',
+      }),
+      // Note that file2 does not contain any changes.
+      createInternalVersionEntry({
+        ...fakeVersionEntry,
+        filename: file2,
+        mime_category: 'text',
+        path: file2,
+        status: '',
+      }),
+      createInternalVersionEntry({
+        ...fakeVersionEntry,
+        filename: file3,
+        mime_category: 'text',
+        path: file3,
+        status: 'M',
+      }),
+    ];
+    const version = createVersionWithEntries(entries);
+
+    it('gets the next path which has a diff', () => {
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file1,
+          pathList: [file1, file2, file3],
+          position: RelativePathPosition.next,
+          version,
+        }),
+      ).toEqual(file3);
+    });
+
+    it('gets the previous path which has a diff', () => {
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file3,
+          pathList: [file1, file2, file3],
+          position: RelativePathPosition.previous,
+          version,
+        }),
+      ).toEqual(file1);
+    });
+
+    it('wraps around to the first path on next', () => {
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file3,
+          pathList: [file1, file2, file3],
+          position: RelativePathPosition.next,
```

Comment:
This test has the same problem: you can change it to `.previous` and it still passes because there are only two file paths with diff changes. The `next` test also has the problem.